modules/local-caching-classloader/src/test/java/org/apache/accumulo/classloader/lcc/LocalCachingContextClassLoaderFactoryTest.java

```diff
@@ -0,0 +1,391 @@
+import org.apache.accumulo.classloader.lcc.resolvers.FileResolver;
+import org.apache.accumulo.core.spi.common.ContextClassLoaderFactory.ContextClassLoaderException;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.FsUrlStreamHandlerFactory;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.MiniDFSCluster;
+import org.eclipse.jetty.server.Server;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class LocalCachingContextClassLoaderFactoryTest {
+
+  private static final int MONITOR_INTERVAL_SECS = 5;
+  private static MiniDFSCluster hdfs;
+  private static FileSystem fs;
+  private static Server jetty;
+  private static URL jarAOrigLocation;
+  private static URL jarBOrigLocation;
+  private static URL jarCOrigLocation;
+  private static URL jarDOrigLocation;
+  private static URL localAllContext;
+  private static URL hdfsAllContext;
+  private static URL jettyAllContext;
+
+  @TempDir
+  private static java.nio.file.Path tempDir;
+
+  @BeforeAll
+  public static void beforeAll() throws Exception {
+    String tmp = tempDir.resolve("base").toUri().toString();
+    System.setProperty(Constants.CACHE_DIR_PROPERTY, tmp);
+
+    // Find the Test jar files
+    jarAOrigLocation =
+        LocalCachingContextClassLoaderFactoryTest.class.getResource("/ClassLoaderTestA/TestA.jar");
+    assertNotNull(jarAOrigLocation);
+    jarBOrigLocation =
+        LocalCachingContextClassLoaderFactoryTest.class.getResource("/ClassLoaderTestB/TestB.jar");
+    assertNotNull(jarBOrigLocation);
+    jarCOrigLocation =
+        LocalCachingContextClassLoaderFactoryTest.class.getResource("/ClassLoaderTestC/TestC.jar");
+    assertNotNull(jarCOrigLocation);
+    jarDOrigLocation =
+        LocalCachingContextClassLoaderFactoryTest.class.getResource("/ClassLoaderTestD/TestD.jar");
+    assertNotNull(jarDOrigLocation);
+
+    // Put B into HDFS
+    hdfs = TestUtils.getMiniCluster();
+    URL.setURLStreamHandlerFactory(new FsUrlStreamHandlerFactory(hdfs.getConfiguration(0)));
+
+    fs = hdfs.getFileSystem();
+    assertTrue(fs.mkdirs(new Path("/contextB")));
+    final Path dst = new Path("/contextB/TestB.jar");
+    fs.copyFromLocalFile(new Path(jarBOrigLocation.toURI()), dst);
+    assertTrue(fs.exists(dst));
+    final URL jarBHdfsLocation = new URL(fs.getUri().toString() + dst.toUri().toString());
+
+    // Put C into Jetty
+    java.nio.file.Path jarCParentDirectory = Paths.get(jarCOrigLocation.toURI()).getParent();
+    assertNotNull(jarCParentDirectory);
+    jetty = TestUtils.getJetty(jarCParentDirectory);
+    final URL jarCJettyLocation = jetty.getURI().resolve("TestC.jar").toURL();
+
+    // ContextDefinition with all jars
+    ContextDefinition allJarsDef = createContextDef("all", jarAOrigLocation, jarBHdfsLocation,
```

Comment:
How would you expect a user to create this json?

Reply:
This could be a follow on, would be helpful to create a little main class or script that a user could point at a dir and it spits out some json.  That could be added to readme.